File: mydefault/src/main/java/com/zhongjh/mydefault/MainActivity.java
package com.zhongjh.mydefault;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * @author zhongjh
 * @date 2022/4/24
 */
public class MainActivity extends AppCompatActivity {

    private MainModel mainModel;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mainModel = new MainModel(new MainParameter());
        mainModel.setName("名称");

        ((TextView) findViewById(R.id.tvName)).setText(mainModel.getName());
    }


}

File: myhilt/src/main/java/com/zhongjh/hiltdemo/mvvm/MainActivity.kt
package com.zhongjh.hiltdemo.mvvm

import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import com.zhongjh.hiltdemo.databinding.ActivityMainBinding
import com.zhongjh.hiltdemo.mvvm.hilt.ViewModel
import dagger.hilt.android.AndroidEntryPoint

/**
 * 这是演示一个 仿造 MVVM的例子
 */
@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    private val viewModel: ViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.tvName.append(viewModel.shopBanner.toString())
    }
}